tests/Unit/NumericalAlgorithms/Spectral/Test_Basis.cpp

```diff
@@ -0,0 +1,101 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <array>
+#include <catch.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "NumericalAlgorithms/Spectral/Basis.hpp"
+#include "Parallel/Printf.hpp"
+#include "Utilities/Blas.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+template <Basis::Polynomials Poly, Basis::Quadrature Quad>
+void test_linear_filter_matrix() {
+  for (size_t n = 2; n < 10; ++n) {
+    const Matrix& filter_matrix = Basis::linear_filter_matrix<
+      Poly, Quad>(n);
+    const Matrix& grid_points_to_spectral_matrix =
+        Basis::grid_points_to_spectral_matrix<Poly, Quad>(n);
+    const DataVector& collocation_points = Basis::collocation_points<
+      Poly, Quad>(n);
+    DataVector u(n);
+    for (size_t s = 0; s < n; ++s) {
+      u[s] = exp(collocation_points[s]);
+    }
+    DataVector u_filtered(n);
+    dgemv_('N', n, n, 1.0, filter_matrix.data(), n, u.data(), 1, 0.0,
+           u_filtered.data(), 1);
+    DataVector u_spectral(n);
+    dgemv_('N', n, n, 1.0, grid_points_to_spectral_matrix.data(), n,
+           u_filtered.data(), 1, 0.0, u_spectral.data(), 1);
+    for (size_t s = 2; s < n; ++s) {
+      CHECK(0.0 == approx(u_spectral[s]));
+    }
+  }
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.Numerical.Spectral.Basis.LinearFilterMatrix",
+    "[NumericalAlgorithms][Spectral][Unit]") {
+  SECTION("Legendre-Gauss") {
+    test_linear_filter_matrix<
+      Basis::Polynomials::Legendre, Basis::Quadrature::Gauss>();
+  }
+  SECTION("Legendre-Gauss-Lobatto") {
+    test_linear_filter_matrix<
+      Basis::Polynomials::Legendre, Basis::Quadrature::GaussLobatto>();
+  }
+}
+
+template <Basis::Polynomials Poly, Basis::Quadrature Quad>
+void test_interpolation_matrix(size_t num_pts, int poly_deg) {
+  const DataVector& collocation_points =
+    Basis::collocation_points<Poly, Quad>(num_pts);
+  DataVector u(num_pts);
+  auto func = [poly_deg](const double x) {
+    double func_value = 1.;
+    for (int p = 1; p <= poly_deg; p++) {
+      func_value += pow(x, p);
+    }
+    return func_value;
+  };
+  for (size_t i = 0; i < num_pts; ++i) {
+    u[i] = func(collocation_points[i]);
```

Comment:
`pow` will work on DataVectors, so you can change `func` to take a `const DataVector&` instead of a double, and then `const DataVector u = func(collocation_points)` here.